### DISCUSSION TITLE
SIWave Simulation Scripting

### DISCUSSION THREAD

**Discussion by loganchoi** [ansys/pyaedt#4156] [Q&A]
I am having difficulty understanding the scripting tool specifically for ANSYS SIWave. Do I have to exclusively start an EDB class and run the simulations through them in my scripting code? I have a folder of .siw files that I just want to automatically run SYZ-computation simulations and extract the S-parameters and collect data.

**Comment by terryli1980** [comment]
I have the same question. It seems like we have to export .siw to EDB file in GUI interface then do some automation scripting based on EDB file.
Is there any method to export .siw project to EDB by scrpting?

**Comment by Samuelopez-ansys** [comment]
@ring630 @svandenb-dev

**Comment by hui-zhou-a** [comment]
@loganchoi @terryli1980 the complete workflow is as below
1, import layout database into EDB class. The database can be brd, odb++, mcm, edb, etc.
2, Configure EDB for simulation.
3, Save EDB and load it into HFSS 3DLayout or SIwave for analysis

However, you want have .siw files already. You can use below code to analyze them 

```
from pyaedt.siwave import Siwave

siwave = Siwave(specified_version="2024.1")
siwave.open_project("xxx.siw")
siwave.oproject.ScrRunDcSimulation(1)
```

As you may already noticed, you can call Ironpython methods

**Reply by terryli1980** [reply]
Hi @ring630 

Thanks for help!
With my limited programming knowledge, I couldn't find how to call Ironpython methods by looking at the documentation.
Since the functionality to export .siw to edb is already available in the GUI interface, do you plan to integrate it into EDB or Siwave classes?

**Reply by loganchoi** [reply]
Hi @ring630, 

I'm looking deeper into PyAEDT, and I am getting an error where if I just start an EDB class, it says I am trying to access memory that could be corrupted or is protected and will crash the program. I have attached the image for your reference. I correctly installed pyaedt and have a functional ANSYS tool for SIwave and Electronics Desktop. Is there a reason why I am getting this error, and/or did I forget to potentially install something or set a path?

![EDBIssue](https://github.com/ansys/pyaedt/assets/76533708/9bcc59d7-8c31-4156-9347-39c76095116f)

**Reply by hui-zhou-a** [reply]
@terryli1980 Export EDB from .siw cannot be implemented due to a technical limitation. 
The soluton on Ironpython is a workaround. If you are not using Ironpython, please ignore it. Please export EDB manually, and follow the workflow I mentioned.

**Reply by terryli1980** [reply]
Hi @ring630 

Thanks for letting me know. 
I have some scripts developed with Ironpython and now I'd like to migrate to Pyaedt and add more functionalities.
Your method is helpful to me. Thanks!
